### PR TITLE
Feat/follow unfollow

### PR DESCRIPTION
**Pull request by SHIKHASINGH1506**
1. Added follow user functionality from right suggestion bar
2. Navigate to other user profile page
3. Follow and unfollow from other user's profile page

### CODE REVIEW

File: src/features/profile/components/profile-card/index.jsx [hunk 1]
```diff
@@ -1,28 +1,75 @@
...
     dispatch(openEditProfileModal());
     dispatch(setEditProfileData(user));
   }
+  const userPostCount = posts.filter(post => post.username === user.username).length;
+  const pathDetail = location.pathname;
+
+  const isFollowing = () => followers?.find(u => u.username === user.username);
+
+  const followUnfollowBtn = isFollowing() ? 'Unfollow' : 'Follow';
+  console.log(followUnfollowBtn);
```
Comment (ApurvaSawant11) [inline src/features/profile/components/profile-card/index.jsx:42]: You can remove this if not needed

**Review by ApurvaSawant11 — APPROVED** [review]
Everything looks good to me! Didn't find anything to suggest. You can merge the PR

File: src/component/suggestions/index.jsx [hunk 2]
```diff
@@ -1,4 +1,42 @@
...
+  useEffect(() => {
+    setSuggestion(allUsers
+      .filter(u => u.username !== user.username)
+      .filter(item => !user.following?.find(followingUser => followingUser._id === item._id))
+    );
+  },
+    [user, allUsers]);
+
+  const isFollowing = followUserId => user.followoing?.find(u => u._id === followUserId) ? true : false;
+
```
Comment (rohanmathur91) [inline src/component/suggestions/index.jsx:23]: Good use of optional chaining, also I think there is a typo in `user.followoing`

File: src/component/suggestions/index.jsx [hunk 3]
```diff
@@ -1,4 +1,42 @@
...
+  const followUserHandler = async followUserId => {
+    try {
+      const response = await dispatch(followUnfollowUser({ 
+        token: token,
```
Comment (rohanmathur91) [inline src/component/suggestions/index.jsx:27]: suggestion: try [axios default headers](https://axios-http.com/docs/config_defaults), so no need to pass the header again and again for each network call.

File: src/features/profile/components/profile-card/index.jsx [hunk 4]
```diff
@@ -1,28 +1,75 @@
...
-      username,
-      followers,
-      following,
-      bio,
-      link,
-      country
-    }
-  } = useSelector(authState);
+export const ProfileCard = ({ userData }) => {
+  console.log(userData);
```
Comment (rohanmathur91) [inline src/features/profile/components/profile-card/index.jsx:8]: remove the console log

**Review by rohanmathur91 — APPROVED** [review]
- Code is readable, great work.
- UX: If the user follows all the suggested users then the suggestions become empty, so show some small image or text message saying eg "No suggestions" or something like that for better UX.
- UX: Adding a new post adds the post below in the feed, it should be on top.
- Also these are just suggestions if you want you can ignore them.

**Comment by SHIKHASINGH1506** [comment]
> * Code is readable, great work.
> 
>     * UX: If the user follows all the suggested users then the suggestions become empty, so show some small image or text message saying eg "No suggestions" or something like that for better UX.
> 
>     * UX: Adding a new post adds the post below in the feed, it should be on top.
> 
>     * Also these are just suggestions if you want you can ignore them.

Thank you for all the suggestions. I'll try to achieve them as per the the time.

**Pull request merged**